builders/xcm/xcm-sdk.md

```diff
@@ -0,0 +1,678 @@
+
+### Native Chain Information {: #native-chain-information }
+
+To get information about each of the Moonbeam network's chain information including the chain key, name, WSS endpoint, parachain ID, protocol asset symbols, chain ID, and units per second, you can access the `moonChain` property:
+
+=== "Moonbeam"
+
+    ```js
+    moonbeam.moonChain
+    ```
+
+=== "Moonriver"
+
+    ```js
+    moonriver.moonChain
+    ```
+
+=== "Moonbase Alpha"
+
+    ```js
+    moonbase.moonChain
+    ```
+
+An example of the data contained in the `moonChain` property is as follows:
+
+```js
+moonChain: {
+  key: 'Moonbeam',
+  name: 'Moonbeam',
+  ws: 'wss://wss.api.moonbeam.network',
+  parachainId: 2004,
+  decimals: 18,
+  chainId: 1284,
+  unitsPerSecond: 10000000000000000000n
+}
+```
+
+### Subscribe to Assets Balance Information {: #subscribe }
+
+To subscribe to balance information, you can use the `subscribeToAssetsBalanceInfo` function and pass in the address you want to get the balance for and a callback function to handle the data:
+
+=== "Moonbeam"
+
+    ```js
+    moonbeam.subscribeToAssetsBalanceInfo('INSERT-ADDRESS', cb)
+    ```
+
+=== "Moonriver"
+
+    ```js
+    moonriver.subscribeToAssetsBalanceInfo('INSERT-ADDRESS', cb)
+    ```
+
+=== "Moonbase Alpha"
+
+    ```js
+    moonbase.subscribeToAssetsBalanceInfo('INSERT-ADDRESS', cb)
+    ```
+
+The following example retrieves the balance information for a given account and prints the balance for each of the supported assets to the console:
+
+```js
+const unsubscribe = await moonbeam.subscribeToAssetsBalanceInfo(
+  'INSERT-ADDRESS',
+  (balances) => {
+    balances.forEach(({ asset, balance, origin }) => {
+      console.log(
+        `${balance.symbol}: ${toDecimal(balance.balance, balance.decimals)} (${
+          origin.name
+        } ${asset.originSymbol})`,
+      );
+    });
+  },
+);
+
+unsubscribe();
+```
+
+### Deposit {: #deposit }
+
+To deposit an asset to Moonbeam from another network, you'll have to first build the request using information from the origin chain before you can send the request. To do so, you'll take the following steps:
+
+1. Call the `deposit` function and pass in the asset symbol for the asset to be deposited. This will return a `chains` array containing the asset's origin network information and a `from` function which will be used to continue to build the request
+2. Call the `from` function and pass in the chain key of the origin network, then call `get` and pass in the address of the account on Moonbeam you want to deposit the funds to and pass in the [Polkadot signer](#creating-signers). This will return information about the origin chain asset, the `xc` representation of the asset on Moonbeam, a `send` function which will be used in the next step, and more
```

Comment:
> and pass in the [Polkadot signer](#creating-signers).

It's not necessary should be a signer. It can be just an address from which you want to deposit if a signer was provided in `init()`.

Here we have 3 cases:
1. If you have a Polkadot compatible signer [(extension)](https://polkadot.js.org/docs/extension/cookbook#sign-and-send-a-transaction) 
1.1 You can pass in init `init({ polkadotSigner })` then in `get` the second argument can be just an address.
1.2 If You want to pass a signer in the later stage to `.get()` then you can do. 
```typescript
  .get(
    'INSERT-MOONBEAM-ADDRESS',
    'INSERT-POLKADOT-ADDRESS',
    { polkadotSigner },
  );
```
2. You don't have an extension. E.g. you need to sign in a Node.js script like we do in the example project. You need to use `Keyring` to sign a transaction. As you described above.
```typescript
  .get(
    'INSERT-MOONBEAM-ADDRESS',
    polkadotAccount,
  );

```
The main confusion here could be that `Signer` != `KeyringPair`.

Reply:
Oh ok. I had tried it by passing in a signer to init and then just passing in an address and I believe I had previously gotten an error so thats why I had messaged you in slack about it. But I'll try again... probably just doing something incorrectly on my end

Reply:
Ah ok I see where I misunderstood. Thank you for clarifying! 🙌

Reply:
With `deposit` it's much more confusing than with `withdraw` just because of how Polkadot.js extension and packages work. As well in the future, I will create a React example. So it's more clear.